drivers/display/display_otm8009a.c

```diff
@@ -0,0 +1,941 @@
+	.blanking_off = otm8009a_blanking_off,
+	.write = otm8009a_write,
+	.read = otm8009a_read,
+	.get_framebuffer = otm8009a_get_framebuffer,
+	.set_brightness = otm8009a_set_brightness,
+	.set_contrast = otm8009a_set_contrast,
+	.get_capabilities = otm8009a_get_capabilities,
+	.set_pixel_format = otm8009a_set_pixel_format,
+	.set_orientation = otm8009a_set_orientation,
+};
+
+static int otm8009a_init(const struct device *dev)
+{
+	const struct otm8009a_config *cfg = dev->config;
+	struct otm8009a_data *data = dev->data;
+	struct mipi_dsi_device mdev;
+	int ret;
+
+	if (cfg->reset.port) {
+		ret = gpio_pin_configure_dt(&cfg->reset, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Reset display failed! (%d)", ret);
+			return ret;
+		}
+		k_msleep(20);
+		ret = gpio_pin_set_dt(&cfg->reset, 1);
+		if (ret < 0) {
+			LOG_ERR("Enable display failed! (%d)", ret);
+			return ret;
+		}
+		k_msleep(10);
+	}
+
+	/* store x/y resolution & rotation */
+	if (cfg->rotation == 0U) {
+		data->xres = cfg->width;
+		data->yres = cfg->height;
+		data->orientation = DISPLAY_ORIENTATION_NORMAL;
+	} else if (cfg->rotation == 90U) {
+		data->xres = cfg->height;
+		data->yres = cfg->width;
+		data->orientation = DISPLAY_ORIENTATION_ROTATED_90;
+	} else if (cfg->rotation == 180U) {
+		data->xres = cfg->width;
+		data->yres = cfg->height;
+		data->orientation = DISPLAY_ORIENTATION_ROTATED_180;
+	} else if (cfg->rotation == 270U) {
+		data->xres = cfg->height;
+		data->yres = cfg->width;
+		data->orientation = DISPLAY_ORIENTATION_ROTATED_270;
+	}
+
+	/* attach to MIPI-DSI host */
+	mdev.data_lanes = cfg->data_lanes;
+	mdev.pixfmt = data->dsi_pixel_format;
+	mdev.mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST | MIPI_DSI_MODE_LPM;
+
+	mdev.timings.hactive = data->xres;
+	mdev.timings.hbp = OTM8009A_HBP;
+	mdev.timings.hfp = OTM8009A_HFP;
+	mdev.timings.hsync = OTM8009A_HSYNC;
+	mdev.timings.vactive = data->yres;
+	mdev.timings.vbp = OTM8009A_VBP;
+	mdev.timings.vfp = OTM8009A_VFP;
+	mdev.timings.vsync = OTM8009A_VSYNC;
+
+	ret = mipi_dsi_attach(cfg->mipi_dsi, cfg->channel, &mdev);
+	if (ret < 0) {
+		LOG_ERR("MIPI-DSI attach failed! (%d)", ret);
+		return ret;
+	}
+
+	ret = otm8009a_check_id(dev);
+	if (ret) {
+		LOG_ERR("Panel ID check failed! (%d)", ret);
+		return ret;
+	}
+
+	ret = otm8009a_configure(dev);
+	if (ret) {
+		LOG_ERR("DSI init sequence failed! (%d)", ret);
+		return ret;
+	}
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_configure_dt(&cfg->backlight, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not configure bl GPIO (%d)", ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+#define OTM8009A_DEVICE(id)									\
+	static const struct otm8009a_config otm8009a_config_##id = {				\
+		.mipi_dsi = DEVICE_DT_GET(DT_INST_BUS(id)),					\
+		.reset = GPIO_DT_SPEC_INST_GET_OR(id, reset_gpios, {0}),			\
+		.backlight = GPIO_DT_SPEC_INST_GET_OR(id, bl_gpios, {0}),			\
+		.data_lanes = DT_INST_PROP_BY_IDX(id, data_lanes, 0),				\
```

Comment:
default doesn't make sense, how is MIPI-DSI driven with 0 data lanes?

Reply:
:thinking: 
That is the index not the default, the property is given (`mipi-dsi-device.yaml`), it is an array and it is mandatory (without default).
I think I can leave it as it is similar in `display_rm68200.c`.
Do you agree?